src/components/LogoutListener/helpers/useNativeAuthLogout.ts

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useRef } from 'react';
+import BigNumber from 'bignumber.js';
+import { useGetAccountInfo } from 'hooks';
+import { useSelector } from 'reduxStore/DappProviderContext';
+import { loginInfoSelector } from 'reduxStore/selectors';
+import {
+  getTokenExpiration,
+  getTokenExpirationTime
+} from 'services/nativeAuth/methods';
+import { logout } from 'utils/logout';
+import { addNewCustomToast, storage } from 'utils';
+import { faRefresh } from '@fortawesome/free-solid-svg-icons';
+import { localStorageKeys } from 'utils/storage/local';
+
+import { TIME_TO_SHOW_WARNING_BEFORE_LOGOUT } from '../constants';
+
+export const useNativeAuthLogout = () => {
+  const { address } = useGetAccountInfo();
+  const { tokenLogin } = useSelector(loginInfoSelector);
+
+  const {
+    isExpired: isNativeAuthTokenExpired,
+    secondsUntilExpires,
+    expiresAt
+  } = getTokenExpiration(tokenLogin?.nativeAuthToken);
+
+  const plannedLogoutRef = useRef('');
+  const logoutTimeoutRef = useRef<NodeJS.Timeout>();
```

Comment:
Oare aici e nevoie de useRef? Nu e suficient o simpla variabila? 

```useEffect(() => {
  const timeout = setTimeout(() => {
    console.log('This will be called after 2 seconds');
  }, 2000);

  return () => clearTimeout(timeout);
}, []);
```

Reply:
There's no other way to do it since we clear the timeout before it's being assigned, if it exists, and it doesn't allow us since TypeScript doesn't recognize it.